### PR TITLE
Update ModelContextProtocol version, early support for new MCP spec

### PR DESCRIPTION
**Pull request by alzimmermsft**
## What does this PR do?

Updates `ModelContextProtocol` dependencies to the latest releases. Adds preliminary support for MCP 2026-07-28 spec version by handling request `_meta`-based client info.

## GitHub issue number?
`[Link to the GitHub issue this PR addresses]`

## Pre-merge Checklist
- [x] Required for All PRs
    - [x] **Read [contribution guidelines](https://github.com/microsoft/mcp/blob/main/CONTRIBUTING.md)**
    - [x] PR title clearly describes the change
    - [x] Commit history is clean with descriptive messages ([cleanup guide](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md))
    - [x] Added comprehensive tests for new/modified functionality
    - [ ] Created a changelog entry if the change falls among the following: new feature, bug fix, UI/UX update, breaking change, or updated dependencies. Follow [the changelog entry guide](https://github.com/microsoft/mcp/blob/main/docs/changelog-entries.md)
- [ ] For MCP tool changes:
    - [ ] **One tool per PR**: This PR adds or modifies only one MCP tool for faster review cycles
    - [ ] Updated `servers/Azure.Mcp.Server/README.md` and/or `servers/Fabric.Mcp.Server/README.md` documentation
    - [ ] Validate `README.md` changes running the script `./eng/scripts/Process-PackageReadMe.ps1`. See [Package README](https://github.com/microsoft/mcp/blob/main/CONTRIBUTING.md#package-readme)
    - [ ] For new or modified tool descriptions, ran [`ToolDescriptionEvaluator`](https://github.com/microsoft/mcp/blob/main/eng/tools/ToolDescriptionEvaluator/Quickstart.md) and obtained a score of `0.4` or more and a top 3 ranking for all related test prompts
    - [ ] For tools with new names, including new tools or renamed tools, update [`consolidated-tools.json`](https://github.com/microsoft/mcp/blob/main/servers/Azure.Mcp.Server/src/Resources/consolidated-tools.json)
    - [ ] For **renamed** tools, follow the [Tool Rename Checklist](https://github.com/microsoft/mcp/blob/main/docs/tool-rename-checklist.md) and tag the PR with the `breaking-change` label
    - [ ] For new tools associated with Azure services or publicly available tools/APIs/products, add URL to documentation in the PR description
- [ ] Extra steps for **Azure MCP Server** tool changes:
    - [ ] Updated command list in [`servers/Azure.Mcp.Server/docs/azmcp-commands.md`](https://github.com/microsoft/mcp/blob/main/servers/Azure.Mcp.Server/docs/azmcp-commands.md)
    - [ ] Ran `./eng/scripts/Update-AzCommandsMetadata.ps1` to update tool metadata in [`azmcp-commands.md`](https://github.com/microsoft/mcp/blob/main/servers/Azure.Mcp.Server/docs/azmcp-commands.md) (required for CI)
    - [ ] Updated test prompts in [`servers/Azure.Mcp.Server/docs/e2eTestPrompts.md`](https://github.com/microsoft/mcp/blob/main/servers/Azure.Mcp.Server/docs/e2eTestPrompts.md)
    - [ ] 👉 For Community (non-Microsoft team member) PRs:
        - [ ] **Security review**: Reviewed code for security vulnerabilities, malicious code, or suspicious activities before running tests (`crypto mining, spam, data exfiltration, etc.`)
        - [ ] **Manual tests run**: added comment `/azp run mcp - pullrequest - live` to run *Live Test Pipeline*

### CODE REVIEW

**Review by vcolin7 — COMMENTED** [review]
I don't have all the context I'd like on how we handle telemetry, but the changes look good at first glance. I'm just curious about the things Copilot flagged :)

File: core/Microsoft.Mcp.Core/src/Services/Telemetry/ITelemetryService.cs [hunk 1]
```diff
@@ -21,9 +21,10 @@ public interface ITelemetryService : IDisposable
     /// </summary>
     /// <param name="activityName">Name of the activity.</param>
     /// <param name="clientInfo">The MCP client information to add to the activity.</param>
+    /// <param name="requestParams">The request parameters for the MCP call. Starting in MCP 2026-07-28 spec, this contains MCP client info.</param>
     /// <returns>An Activity object or null if there are no active listeners or telemetry is disabled.</returns>
     /// <exception cref="InvalidOperationException">If the service is not in an operational state or <see cref="InitializeAsync"/> was not invoked.</exception>
-    Activity? StartActivity(string activityName, Implementation? clientInfo);
+    Activity? StartActivity(string activityName, Implementation? clientInfo, RequestParams requestParams);
```
Comment (RickWinter) [inline core/Microsoft.Mcp.Core/src/Services/Telemetry/ITelemetryService.cs]: The interface declares `RequestParams requestParams` as non-nullable, but every caller and impl treats it as nullable: the `StartActivity(name)` overload passes `null`, `TelemetryService` is `RequestParams?`, and `McpRuntime` passes `request.Params` which it null-checks on the very next line. `NoopTelemetryService` also keeps it non-null, so the three signatures disagree. Let's make this `RequestParams?` to match reality and keep the contract honest.

File: core/Microsoft.Mcp.Core/src/Services/Telemetry/TelemetryService.cs [hunk 2]
```diff
@@ -114,6 +111,31 @@ public TelemetryService(IMachineInformationProvider informationProvider,
...
+    internal static void SetClientNameAndVersion(Activity activity, Implementation? clientInfo, RequestParams? requestParams)
+    {
+        if (clientInfo != null)
+        {
+            activity.SetTag(TagName.ClientName, clientInfo.Name)
+                .SetTag(TagName.ClientVersion, clientInfo.Version);
+        }
+
+        if (requestParams?.Meta != null &&
+            requestParams.Meta.TryGetPropertyValue("io.modelcontextprotocol/clientInfo", out var node) &&
```
Comment (RickWinter) [inline core/Microsoft.Mcp.Core/src/Services/Telemetry/TelemetryService.cs]: nit: `"io.modelcontextprotocol/clientInfo"` and the `"name"`/`"version"` keys are well-known spec values duplicated here and again in the tests. Can we hoist them into consts (alongside `TagName`) so the meta key isn't typo-prone and the spec dependency lives in one place?

File: core/Microsoft.Mcp.Core/src/Services/Telemetry/TelemetryService.cs [hunk 3]
```diff
@@ -114,6 +111,31 @@ public TelemetryService(IMachineInformationProvider informationProvider,
...
+            node is JsonObject requestClientInfo)
+        {
+            if (requestClientInfo.TryGetPropertyValue("name", out var nameNode) &&
+                nameNode is JsonValue nameValue)
+            {
+                activity.SetTag(TagName.ClientName, nameValue.GetValue<string>());
```
Comment (RickWinter) [inline core/Microsoft.Mcp.Core/src/Services/Telemetry/TelemetryService.cs]: `GetValue<string>()` throws `InvalidOperationException` when the node is a `JsonValue` that isn't a string (e.g. a client sends `"name": 123`). This runs in `StartActivity` before the try/catch in `CallToolHandler`/`ListToolsHandler`, so malformed client meta would surface as a failed tool call rather than just dropped telemetry. The new tests cover the array/string-at-clientInfo cases but not a non-string `name`/`version`. Can we use `nameValue.TryGetValue<string>(out var n)` (and same for version) and guard on the kind? Same applies to line 134.

Comment (alzimmermsft) [reply core/Microsoft.Mcp.Core/src/Services/Telemetry/TelemetryService.cs]: Now using `TryGetValue`.

File: core/Microsoft.Mcp.Core/src/Services/Telemetry/TelemetryService.cs (hunk 2 above)
Comment (alzimmermsft) [reply core/Microsoft.Mcp.Core/src/Services/Telemetry/TelemetryService.cs]: Added to `McpHelper` as that's a better place for this as it isn't telemetry specific.

File: core/Microsoft.Mcp.Core/src/Services/Telemetry/ITelemetryService.cs (hunk 1 above)
Comment (alzimmermsft) [reply core/Microsoft.Mcp.Core/src/Services/Telemetry/ITelemetryService.cs]: Updated to make the interface take `RequestParams?`

File: core/Microsoft.Mcp.Core/src/Services/Telemetry/ITelemetryService.cs [hunk 4]
```diff
@@ -21,9 +21,10 @@ public interface ITelemetryService : IDisposable
     /// </summary>
     /// <param name="activityName">Name of the activity.</param>
     /// <param name="clientInfo">The MCP client information to add to the activity.</param>
+    /// <param name="requestParams">The request parameters for the MCP call. Starting in MCP 2026-07-28 spec, this contains MCP client info.</param
```
Comment (jongio) [inline core/Microsoft.Mcp.Core/src/Services/Telemetry/ITelemetryService.cs:24]: Nit: this `</param` tag is missing the closing `>`. Should be `</param>`.

**Review by jongio — COMMENTED** [review]
Dependency bump from ModelContextProtocol 1.1.0 to 1.4.0 looks solid. The new SetClientNameAndVersion logic correctly handles the per-request _meta-based client info with proper precedence (meta overrides session-level Implementation per-field) and uses TryGetValue<string> for type safety. Tests cover the edge cases well (partial fields, non-string values, arrays, missing keys).

One minor doc nit below, otherwise this is ready to go.
